Task3/src/Data/App.hs

```diff
@@ -0,0 +1,54 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# OPTIONS_GHC -Wno-partial-fields #-}
+
+module Data.App (App(..), AppResult(..), AppConfig(..), AppState(..), AppError(..), AppLog, start) where
+
+import Control.Monad.IO.Class
+import Control.Monad.Reader
+import Control.Monad.State
+import Control.Monad.Writer
+import Data.Entities ( Customer, Order, Product, Shop, ProductOrder )
+import Control.Monad.Except 
+
+newtype App a = App {
+        runApp :: ExceptT AppError (WriterT AppLog (ReaderT AppConfig (StateT AppState IO))) a
+    } deriving (Functor, Applicative, Monad,
+                MonadIO,
+                MonadReader AppConfig,
+                MonadState AppState,
+                MonadWriter AppLog,
+                MonadError AppError)
+
+data AppResult a = AppResult { appResult :: a }
+                 | AppError { message :: String }
+                deriving (Show)
```

Comment:
Let's "unwrap" all the data collected by the app into AppResult in case of success (and some data in case of error too): logs, current state and configuration